### PR TITLE
docs: UI refactor ADR

### PR DESCRIPTION
**Pull request by nsprenkle**
**TL;DR -** Add ADR for UI mixin refactors. 

**What changed?**

- Add ADR 003
- Link repo docs in README.rst

**Developer Checklist**

- [x] Reviewed the [release process](https://github.com/openedx/edx-ora2/blob/master/.github/release_process.md)
- [ ] Bumped version number in [openassessment/\_\_init\_\_.py](https://github.com/openedx/edx-ora2/blob/master/openassessment/__init__.py#L4) and [package.json](https://github.com/openedx/edx-ora2/blob/master/package.json#L3)

**Testing Instructions**

n/a

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual documentation review

FYI: @openedx/content-aurora

### CODE REVIEW

File: docs/decisions/0003-ui-mixins.rst [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+presentation logic in complicated ways. This made refactoring and accessing that
+underlying data in any other context challenging.
+
+Along the way to creating a new ORA UI, we decided on some refactoring to aid access
+to that data and allow us to reuse that code across UI implementations.
+
+Decisions
+*********
+
+- We will standalone APIs for accessing data on an ORA block, placed in
```
Comment (mattcarter) [inline docs/decisions/0003-ui-mixins.rst:25]: missing word?  

> - We will ***_create_*** standalone APIs...

File: docs/decisions/0003-ui-mixins.rst [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+- New UI variants will also be created in ``openassessment/xblock/ui_mixins`` as separate
+  folders.
+
+Consequences
+************
+
+Standalone data / utility APIs
+==============================
+
+Splitting the data / utilities out of existing mixins and into a standalone location
```
Comment (mattcarter) [inline docs/decisions/0003-ui-mixins.rst:39]: Suggest:
> standalone location -> separate locations

File: docs/decisions/0003-ui-mixins.rst [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+allows us to access that data across multiple contexts, without having to duplicate
+business logic.
+
+These APIs, in addition to serving our legacy UI views can also be leveraged in future
```
Comment (mattcarter) [inline docs/decisions/0003-ui-mixins.rst:43]: nit: 
adjectival clause needs a trailing comma.

> These APIs, in addition to serving our legacy UI views, can ...

**Review by mattcarter — APPROVED** [review]
Approved.
